### PR TITLE
Revealer: Use absolute units for the fixed background

### PR DESCRIPTION
**Pull request by regiskuckaertz**
It turns out that using % to set the dimensions of a fixed positioned element is a bad idea, especially if this element has a background image. Even using the shiny `vh` and `vw` units doesn't provide smooth scrolling.

I had to instead use absolute units and set the dimensions in pixels once and for all, at runtime.

cc  @guardian/commercial-dev

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js [hunk 1]
```diff
@@ -24,11 +25,14 @@ define([
                 }
             }).then(function () {
                 return fastdom.read(function () {
-                    return $adSlot[0].getBoundingClientRect();
+                    return detect.getViewport();
                 });
-            }).then(function (adSlotRect) {
+            }).then(function (viewport) {
                 var background = $adSlot[0].getElementsByClassName('creative__background')[0];
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: Shouldn't you do that in the `fastdom.read()` method?

File: static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js [hunk 2]
```diff
@@ -24,11 +25,14 @@ define([
...
-                background.style.left = (adSlotRect.left + adSlotRect.width / 2) + 'px';
+                background.style.width = viewport.width + 'px';
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: Use `fastdom.write()`?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: Damn. All these callbacks. I didn't see it. Thanks!

**Comment by regiskuckaertz** [comment]
@Calanthe done 😄

File: static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js [hunk 3]
```diff
@@ -24,12 +25,17 @@ define([
...
+                    return detect.getViewport();
+                });
+            }).then(function (viewport) {
+                return fastdom.write(function () {
+                    var background = $adSlot[0].getElementsByClassName('creative__background')[0];
+                    background.style.width = viewport.width + 'px';
+                    // for the height, we need to account for the height of the location bar, which
+                    // may or may not be there. 70px padding is not too much.
+                    background.style.height = (viewport.height + 70) + 'px';
+                    return true;
```
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: I am not sure if I understand how this works. Why do you need to `return true`? Is this method called in here https://github.com/guardian/frontend/blob/2a4f00ba2e7c5b7abd1fff614b03cd7cd4947fea/static/src/javascripts/projects/common/modules/commercial/dfp/private/apply-creative-template.js#L104 ?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: Yes, the returned value is important to let our code know the rendering went correctly.

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: [Here](https://github.com/guardian/frontend/blob/2a4f00ba2e7c5b7abd1fff614b03cd7cd4947fea/static/src/javascripts/projects/common/modules/commercial/dfp/private/Advert.js#L75)

Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js]: And then it is resolved. Now I got it ;)

**Comment by Calanthe** [comment]
👍

**Comment by TBonnin** [comment]
Hi @regiskuckaertz I have reverted this PR last night since we needed to deploy and it was merged but not pushed to prod.

Feel free to merge it again.

**Comment by regiskuckaertz** [comment]
@TBonnin I pushed static to PROD, I thought that was all that was necessary. Sorry about that.

**Pull request merged**